### PR TITLE
chore: update tslint and codelyzer

### PR DESCRIPTION
**Pull request by mgechev**
1. Update to tslint ^4.0.2.
2. Update to codelyzer 2.0.0-beta.1 which includes external templates & styles support.
3. Drop deprecated tslint rules.
4. Update to the latest rule changes in codelyzer.

### CODE REVIEW

File: packages/angular-cli/blueprints/ng2/files/tslint.json [hunk 1]
```diff
@@ -93,12 +90,8 @@
       "check-type"
     ],
 
-    "directive-selector-prefix": [true, "<%= prefix %>"],
-    "component-selector-prefix": [true, "<%= prefix %>"],
-    "directive-selector-name": [true, "camelCase"],
-    "component-selector-name": [true, "kebab-case"],
-    "directive-selector-type": [true, "attribute"],
-    "component-selector-type": [true, "element"],
+    "directive-selector": [true, "attribute", "<%= prefix %>", "camelCase"],
```
Comment (mgechev) [inline packages/angular-cli/blueprints/ng2/files/tslint.json]: Note that a list of prefixes could be used, i.e.:

```
 "directive-selector": [true, "attribute", ["<%= prefix %>", "bar", "baz"], "camelCase"]
```

**Comment by mgechev** [comment]
Hm, travis and appveyor are unhappy with the change.

File: packages/angular-cli/blueprints/ng2/files/tslint.json [hunk 2]
```diff
@@ -109,6 +101,8 @@
     "component-class-suffix": true,
     "directive-class-suffix": true,
     "templates-use-public": true,
+    "no-access-missing-member": true,
+    "templates-use-public": true,
```
Comment (deebloo) [inline packages/angular-cli/blueprints/ng2/files/tslint.json]: "templates-use-public" is defined twice

Comment (mgechev) [inline packages/angular-cli/blueprints/ng2/files/tslint.json]: Thanks!

**Comment by hansl** [comment]
LGTM.

**Pull request merged**